src/piggui.rs

```diff
@@ -238,7 +355,80 @@ impl Application for Piggui {
             .center_x()
             .center_y();
 
-        self.toast_handler.view(content.into())
+        if self.show_modal {
```

Comment:
We should move all of this to a view() method of a ConnectDialog (or similar) module...